### PR TITLE
MRT core is restoring the packages referenced in WindowsAppSDKInsight…

### PR DESCRIPTION
**Pull request by ZenBird**
…s project. Since the project contains a reference to Microsoft.Telemetry.Inbox.Native this restore will fail causing the MRT checkins to fail. We must not restore

Microsoft.Telemetry.Inbox.Native during CI builds. The following steps have been taken to address this issue

* Remove reference to Microsoft.Telemetry.Inbox.Native from WindowsAppSDKInsightsLib project
* Create a new packages.config that contains a single reference to Microsoft.Telemetry.Inbox.Native.
* Update the build script to restore the package using the new config file

### CODE REVIEW

**Review by Mmrad07 — COMMENTED** [review]
👍

**Review by Mmrad07 — APPROVED** [review]

**Comment by huichen123** [comment]
> #    displayName: 'Replace existing stubbed MicrosoftTelemetry.h header with the real version from the nuget package'

does this need to be enabled?

---
Refers to: build/ProjectReunion-BuildFoundation.yml:100 in fd9b8b2. [](commit_id = fd9b8b2ebcfb293e9a4a6c429b01aaf514fd263e, deletion_comment = False)

**Comment by huichen123** [comment]
nuget.config still has the private feed. I am surprised it's not detected by the system. I think it's not allowed. @kythant

**Comment by ZenBird** [comment]
> #    displayName: 'Replace existing stubbed MicrosoftTelemetry.h header with the real version from the nuget package'

This is code that existed before. I did not want to remove it in-case someone was parking it here do something later. The code that does the replace it above line 73-82 and in build-mrt.yml line 97-109

---
In reply to: [899765785](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899765785) [](http://example.com/codeflow?ancestors=899765785,899765785)

---
Refers to: build/ProjectReunion-BuildFoundation.yml:100 in fd9b8b2. [](commit_id = fd9b8b2ebcfb293e9a4a6c429b01aaf514fd263e, deletion_comment = False)

**Comment by ZenBird** [comment]
That is because this is not run by the CI build.

---
In reply to: [899767260](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899767260) [](http://example.com/codeflow?ancestors=899767260,899767260)

**Comment by huichen123** [comment]
> > # displayName: 'Replace existing stubbed MicrosoftTelemetry.h header with the real version from the nuget package'
> 
> This is code that existed before. I did not want to remove it in-case someone was parking it here do something later. The code that does the replace is in lines 73-82 and in build-mrt.yml line 97-109
> 
> In reply to: [899765785](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899765785)
> 
> Refers to: build/ProjectReunion-BuildFoundation.yml:100 in [fd9b8b2](https://github.com/microsoft/WindowsAppSDK/commit/fd9b8b2ebcfb293e9a4a6c429b01aaf514fd263e). [](commit_id = [fd9b8b2](https://github.com/microsoft/WindowsAppSDK/commit/fd9b8b2ebcfb293e9a4a6c429b01aaf514fd263e), deletion_comment = False)

doing it in build-mrt.yml is probably not appropriate. build-mrt.yml runs in both public PR pipeline and internal pipeline. Stub replacement should only be done in internal build.

**Comment by huichen123** [comment]
that's a different issue. not permission, but security hole.

---
In reply to: [899768676](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899768676) [](http://example.com/codeflow?ancestors=899768676,899767260)

**Comment by ZenBird** [comment]
The code in build-mrt.yml searches for "MicrosoftTelemetry.h" and replaces "TraceloggingConfig.h" with "MSTelemetry.h". Running build-mrt.yml alone will skip this step since there will be no mstelemetry.h when run in CI builds.


---
In reply to: [899769855](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899769855) [](http://example.com/codeflow?ancestors=899769855,899769855)

**Comment by ZenBird** [comment]
Can you elaborate please? My understanding is that ProjectReunion-BuildFoundation.yml is run in official builds and not in CI. Even if someone were to edit build-mrt.yml and dump "MsTelemetry.h", it will not do anything since the code is not running as part of the official build.

---
In reply to: [899771052](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899771052) [](http://example.com/codeflow?ancestors=899771052,899767260)

**Comment by huichen123** [comment]
[https://www.1eswiki.com/wiki/How_to_Securely_Configure_Package_Source_Files](https://www.1eswiki.com/wiki/How_to_Securely_Configure_Package_Source_Files)

---
In reply to: [899778414](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899778414) [](http://example.com/codeflow?ancestors=899778414,899767260)

**Comment by ZenBird** [comment]
Pending: Updated the nuget.config to reference only the internal feed and no others.

---
In reply to: [899781571](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899781571) [](http://example.com/codeflow?ancestors=899781571,899767260)

File: dev/WindowsAppSDK_Insights/packages.config [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.Telemetry.Inbox.Native" version="10.0.19041.1-191206-1406.vb-release.x86fre" targetFramework="native" />
```
Comment (huichen123) [inline dev/WindowsAppSDK_Insights/packages.config:3]: >x86fre [](http://example.com/codeflow?start=94&length=6)

i'd assume this is ok assuming it's header only? but want to get confirmation.

**Comment by huichen123** [comment]
>       <clear />

nit: extra 2 spaces

---
Refers to: dev/WindowsAppSDK_Insights/nuget.config:8 in 2eff642. [](commit_id = 2eff64239d7d3a319de9a4e72b85706ce1e65d00, deletion_comment = False)

File: dev/WindowsAppSDK_Insights/packages.config (hunk 1 above)
Comment (ZenBird) [reply dev/WindowsAppSDK_Insights/packages.config:3]: That is correct. This is the latest package that is published.

**Review by huichen123 — APPROVED** [review]

**Comment by ZenBird** [comment]
> #    displayName: 'Replace existing stubbed MicrosoftTelemetry.h header with the real version from the nuget package'

If no one has any objections, I will remove this dead code in the next iteration.

---
In reply to: [899768044](https://github.com/microsoft/WindowsAppSDK/pull/1243#issuecomment-899768044) [](http://example.com/codeflow?ancestors=899768044,899765785,899765785)

---
Refers to: build/ProjectReunion-BuildFoundation.yml:100 in fd9b8b2. [](commit_id = fd9b8b2ebcfb293e9a4a6c429b01aaf514fd263e, deletion_comment = False)

**Review by kythant — APPROVED** [review]

**Pull request merged**